nixos/modules/security/acme/default.nix

```diff
@@ -428,6 +428,34 @@ let
 
   certConfigs = mapAttrs certToConfig cfg.certs;
 
+  accountGroupedCerts = (groupBy (conf: conf.accountHash) (attrValues certConfigs));
+
+  # Concurrency control for renewal services.
+  # L = maxConcurrentRenewals
+  # Every Nth cert is "paired" with an After= condition
+  # with the N+Lth cert in front of it.
+  # If <= N certs are configured, no limiting is applied.
+  # Leader certs for a given account target are excluded from the logic.
+  maxConcurrentRenewals = cfg.maxConcurrentRenewals;
+  certNames = builtins.attrNames certConfigs;
+  certServices = builtins.listToAttrs (
+    imap0 (i: cert: let
+      iPaired = i + maxConcurrentRenewals;
+      pairedCert = builtins.elemAt certNames iPaired;
+      conf = certConfigs."${cert}";
+      # Same logic for leader as in accountTargets definition
+      leader = builtins.head accountGroupedCerts."${conf.accountHash}";
+    in {
+      name = "acme-${cert}";
+      value = mkMerge [
+        conf.renewService
+        (optionalAttrs (iPaired < numCerts && leader.cert != cert) {
```

Comment:
Do we require a setting of `maxConcurrentRenewals` that disables concurrency limits altogether? When setting `maxConcurrentRenewals = 0`, which achieves this behaviour in #244511, each service receives a `After=` relationship towards itself.
While the services still appear to run fine and now have no concurrency limit, this is a bit awkward.

Reply:
Ah yes that's a bug. 0 should result in no relations